prometheus/promhttp/http.go

```diff
@@ -52,7 +52,21 @@ const (
 	acceptEncodingHeader  = "Accept-Encoding"
 )
 
-var bufPool sync.Pool
+var (
+	bufPool sync.Pool
+
+	scrapesCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "promhttp_metric_handler_requests_total",
+			Help: "Total number of scrapes by HTTP status code.",
+		},
+		[]string{"code"},
```

Comment:
If you are going to use a label, initialise the potential label values. Personally I'd do this without a label.

Reply:
The label is important here, partially because of upcoming features like limited number of requests in flight.
We cannot really initialize all potential status codes here. Would you prefer to initialize a few common ones?

Reply:
There's a few we expect to see, like 200 and whatever the duplicate labels one is.

Reply:
> There's a few we expect to see, like 200 and whatever the duplicate labels one is.

I'm well aware of that, and it wasn't my point. My point was if it makes sense to initialize a few (but not all) label values. (This is a true question, where I want to know your opinion.)

Reply:
Ah yes, it makes sense to initialise what we know. That will avoid most of the potential breakage.